newlib/libc/machine/riscv/memcpy-asm.S

```diff
@@ -28,5 +28,91 @@ memcpy:
 2:
   ret
 
-  .size	memcpy, .-memcpy
+#else
+.Lwordify:
+    andi   a3, a0, 3
+    beqz   a3, .Laligned
+    li    a3, 3
+    bleu  a2, a3, .Lcopy1
+
+    lbu    a3, 0(a1)
+    sb     a3, 0(a0)
+    addi   a0, a0, 1
+    addi   a1, a1, 1
+    addi   a2, a2, -1
+    j      .Lwordify
+
+.Laligned:
+#ifdef __riscv_misaligned_avoid
+    andi   a3, a1, 3
+    bnez   a3, .Lcopy1
 #endif
+
+.Lcopy32:
+    li     a3, 32
+    blt    a2, a3, .Lcopy16
+    lw     t1, 0(a1)
+    lw     t2, 4(a1)
+    lw     t3, 8(a1)
```

Comment:
this does not work with rv32e. t3 is not contained in rv32e. If you want to have one assembly version for RISC-V, you need to work with x0-x15 only.